src/Microsoft.AspNet.OData/Routing/ODataActionSelector.cs

```diff
@@ -106,5 +112,80 @@ private static HttpResponseMessage CreateErrorResponse(HttpRequestMessage reques
 
             return request.CreateErrorResponse(statusCode, error);
         }
+
+        private static bool ActionParametersMatchRequest(HttpActionDescriptor action, HttpControllerContext context)
+        {
+            var parameters = action.GetParameters();
+            var routeData = context.RouteData;
+            var conventionsStore = context.Request.ODataProperties().RoutingConventionsStore;
+            var matchedBody = false;
+            var route = routeData.Route as ODataRoute;
+            var routePrefix = route?.RoutePrefix;
+            var availableKeys = routeData.Values.Keys
+                .Where(k => routePrefix != "{" + k + "}"
+                    && k != ODataRouteConstants.Action
+                    && k != ODataRouteConstants.Controller
+                    && k != ODataRouteConstants.ODataPath)
```

Comment:
This logic seems a bit brittle, given that you do the check on line 132 to make sure that you have eliminated any "special" route values if the action has no parameters.  If someone were to add another special-purpose route value in the future, then this would have to be updated to omit it.  Is this line, plus line 132, just an optimization? If all of the parameters for the action are accounted for, should we care if there are other route values?